src/mutation.cpp

```diff
@@ -835,28 +835,30 @@ void player::mutate_towards(std::string mut)
         mutation_effect(*this, mut);
         mutation_replaced = true;
     }
-    if (canceltrait != "") {
-        if(traits[mut].mixed_effect || traits[canceltrait].mixed_effect) {
-            rating = m_mixed;
-        } else if(traits[mut].points <= 0 && traits[canceltrait].points > 0) {
-            rating = m_good;
-        } else if(traits[mut].points > 0 && traits[canceltrait].points <= 0) {
-            rating = m_bad;
-        } else if(traits[mut].points == 0 && traits[canceltrait].points == 0) {
-            rating = m_neutral;
-        } else {
-            rating = m_mixed;
+    if (canceltrait.size() > 0) {
```

Comment:
OK.  I'm concerned with making sure the loop gets run if there's anything in the vector, not making sure it isn't run if the vector is empty, so I take it that just leaving the loop in and removing the check would be fine.